### PR TITLE
[FLINK-23109]Translate "Scala API Extensions" pages into Chinese

### PR DESCRIPTION
**Pull request by zhangzhl5**
## What is the purpose of the change

Translate "Scala API Extensions" pages into Chinese.

The page url is https://ci.apache.org/projects/flink/flink-docs-release-1.13/zh/docs/dev/datastream/scala_api_extensions/

## Brief change log

- *translate flink/docs/content.zh/docs/dev/datastream/scala_api_extensions.md*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no 

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 1]
```diff
@@ -25,29 +25,23 @@ specific language governing permissions and limitations
...
+# Scala API扩展
 
-In order to keep a fair amount of consistency between the Scala and Java APIs, some
-of the features that allow a high-level of expressiveness in Scala have been left
-out from the standard APIs for both batch and streaming.
+为了在Scala和Java api之间保持一定的一致性，在批处理和流处理的标准api中省略了一些允许Scala高级表达的特性。
 
-If you want to _enjoy the full Scala experience_ you can choose to opt-in to
-extensions that enhance the Scala API via implicit conversions.
+如果你想体验完整的Scala，可以选择通过隐式转换增强Scala API的扩展。
```
Comment (95chenjz) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:32]: ”如果你想体验完整的Scala“ -- ”如果你想 _拥有完整的 Scala 体验_“

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 2]
```diff
@@ -184,22 +176,21 @@ data1.join(data2).
 
 
 
-For more information on the semantics of each method, please refer to the [DataStream]({{< ref "docs/dev/datastream/overview" >}}) API documentation.
+有关每个方法语义的更多信息, 请参考 DataStream API 文档。
```
Comment (95chenjz) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:179]: Data Stream link missed.

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 3]
```diff
@@ -25,29 +25,23 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-# Scala API Extensions
+# Scala API扩展
 
-In order to keep a fair amount of consistency between the Scala and Java APIs, some
-of the features that allow a high-level of expressiveness in Scala have been left
-out from the standard APIs for both batch and streaming.
+为了在Scala和Java api之间保持一定的一致性，在批处理和流处理的标准api中省略了一些允许Scala高级表达的特性。
```
[the remark this answers is not in the document]
Comment (zhangzhl5) [reply docs/content.zh/docs/dev/datastream/scala_api_extensions.md]: API 已统一为大写

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md (hunk 2 above)
Comment (zhangzhl5) [reply docs/content.zh/docs/dev/datastream/scala_api_extensions.md]: Data Stream link re-added.

**Comment by zhangzhl5** [comment]
hi， @wuchong  I have modified it according to the suggestion,could you help to review this? thanks

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Scala API Extensions"
+title: "Scala API扩展"
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:2]:
```suggestion
title: "Scala API 扩展"
```

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 5]
```diff
@@ -25,29 +25,23 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-# Scala API Extensions
+# Scala API扩展
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:28]:
```suggestion
# Scala API 扩展
```

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 6]
```diff
@@ -25,29 +25,23 @@ specific language governing permissions and limitations
...
 
-In order to keep a fair amount of consistency between the Scala and Java APIs, some
-of the features that allow a high-level of expressiveness in Scala have been left
-out from the standard APIs for both batch and streaming.
+为了在 Scala 和 Java API 之间保持一定的一致性，在批处理和流处理的标准 API 中省略了一些允许 Scala 高级表达的特性。
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:30]: What about `为了在 Scala 和 Java API 之间保持一定程度的一致性，在批处理和流处理的标准 API 中舍弃了一些 Scala 高级表达式的特性。`？
It's only a tentative suggestion. Maybe we should translate it in a better way.

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 7]
```diff
@@ -25,29 +25,23 @@ specific language governing permissions and limitations
...
 
-If you want to _enjoy the full Scala experience_ you can choose to opt-in to
-extensions that enhance the Scala API via implicit conversions.
+如果你想拥有完整的 Scala 体验，可以选择通过隐式转换增强 Scala API 的扩展。
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:32]: Maybe we keep `_***_` format in  `如果你想拥有 _完整的 Scala 体验_，可以选择通过隐式转换增强 Scala API 的扩展。`
if shown in correct style.

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 8]
```diff
@@ -25,29 +25,23 @@ specific language governing permissions and limitations
...
-To use all the available extensions, you can just add a simple `import` for the DataStream API
+要使用所有可用的扩展，你只需为 DataStream API 添加一个简单的引入
 
 {{< highlight scala >}}
 import org.apache.flink.streaming.api.scala.extensions._
 {{< /highlight >}}
 
-Alternatively, you can import individual extensions _a-là-carte_ to only use those
-you prefer.
+或者，您可以引入单个扩展 a-là-carte 来使用您喜欢的扩展。
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:40]:
```suggestion
或者，你可以引入单个扩展 _a-là-carte_ 来使用你喜欢的扩展。
```

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 9]
```diff
@@ -25,29 +25,23 @@ specific language governing permissions and limitations
...
 
-## Accept partial functions
+## 接受部分功能
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:42]: IMO, Maybe it's a good choice to keep  original content

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 10]
```diff
@@ -184,22 +176,21 @@ data1.join(data2).
 
 
 
-For more information on the semantics of each method, please refer to the [DataStream]({{< ref "docs/dev/datastream/overview" >}}) API documentation.
+有关每个方法语义的更多信息, 请参考 [DataStream]({{< ref "docs/dev/datastream/overview" >}}) API 文档。
 
-To use this extension exclusively, you can add the following `import`:
+要单独使用此扩展，您可以添加以下引入：
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:181]:
```suggestion
要单独使用此扩展，你可以添加以下引入：
```

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 11]
```diff
@@ -184,22 +176,21 @@ data1.join(data2).
...
 
 {{< highlight scala >}}
 import org.apache.flink.api.scala.extensions.acceptPartialFunctions
 {{< /highlight >}}
 
-for the DataSet extensions and
+用于DataSet扩展
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:187]:
```suggestion
用于 DataSet 扩展
```

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md [hunk 12]
```diff
@@ -184,22 +176,21 @@ data1.join(data2).
...
 
 {{< highlight scala >}}
 import org.apache.flink.streaming.api.scala.extensions.acceptPartialFunctions
 {{< /highlight >}}
 
-The following snippet shows a minimal example of how to use these extension
-methods together (with the DataSet API):
+下面的代码片段展示了如何一起使用这些扩展方法(以及 DataSet API)的最小示例:
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/datastream/scala_api_extensions.md:193]:
```suggestion
下面的代码片段展示了如何一起使用这些扩展方法（以及 DataSet API）的最小示例:
```

**Review by RocMarshal — COMMENTED** [review]
Thank you @zhangzhl5  for your contribution. I left some comments. Please let me know what do you think of it.

File: docs/content.zh/docs/dev/datastream/scala_api_extensions.md (hunk 12 above)
Comment (zhangzhl5) [reply docs/content.zh/docs/dev/datastream/scala_api_extensions.md]: hi @RocMarshal Thank you for your suggestion, I have revised it again, is there anything else I can do？

**Review by RocMarshal — COMMENTED** [review]
Thank you @zhangzhl5 @95chenjz for your effort.  LGTM now. 
And we need a committer to do the final examination.
Hi, @wuchong , could you please check it ? Thank you.

**Review by RocMarshal — APPROVED** [review]
LGTM +1.

**Comment by RocMarshal** [comment]
Hi, @wuchong , The CI is passed.

**Pull request merged**